src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -597,7 +602,9 @@ export class Item extends Events.EventEmitter {
 		var eventData: IItemDisposeEvent = { item: this };
 		this.emit('item:dispose', eventData);
 
-		this.registry.deregister(this);
+		if (this.registry.isRegistered(this.id)) {
+			this.registry.deregister(this);
+		}
```

Comment:
According to ids not being unique it means that if you create two items with same ID they will be registered once and unregistered straight after disposing any of them. Does registry produce event of item disposal? This code hardly affects it.